Index: .idea/calculator.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<module type=\"PYTHON_MODULE\" version=\"4\">\r\n  <component name=\"NewModuleRootManager\">\r\n    <content url=\"file://$MODULE_DIR$\" />\r\n    <orderEntry type=\"inheritedJdk\" />\r\n    <orderEntry type=\"sourceFolder\" forTests=\"false\" />\r\n  </component>\r\n</module>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/calculator.iml b/.idea/calculator.iml
--- a/.idea/calculator.iml	(revision 7079a0bbdeae42e04c8ccd9721fd26cb9d54cd74)
+++ b/.idea/calculator.iml	(date 1766956232499)
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <module type="PYTHON_MODULE" version="4">
   <component name="NewModuleRootManager">
-    <content url="file://$MODULE_DIR$" />
-    <orderEntry type="inheritedJdk" />
+    <content url="file://$MODULE_DIR$">
+      <excludeFolder url="file://$MODULE_DIR$/.venv" />
+    </content>
+    <orderEntry type="jdk" jdkName="Python 3.10 (calculator)" jdkType="Python SDK" />
     <orderEntry type="sourceFolder" forTests="false" />
   </component>
 </module>
\ No newline at end of file
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<project version=\"4\">\r\n  <component name=\"ProjectRootManager\" version=\"2\" project-jdk-name=\"Python 3.10\" project-jdk-type=\"Python SDK\" />\r\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
--- a/.idea/misc.xml	(revision 7079a0bbdeae42e04c8ccd9721fd26cb9d54cd74)
+++ b/.idea/misc.xml	(date 1766956232505)
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.10" project-jdk-type="Python SDK" />
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.10 (calculator)" project-jdk-type="Python SDK" />
 </project>
\ No newline at end of file
Index: Reform/reform.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Reform/reform.py b/Reform/reform.py
new file mode 100644
--- /dev/null	(date 1766956362225)
+++ b/Reform/reform.py	(date 1766956362225)
@@ -0,0 +1,73 @@
+from Exceptions import NumbersInARowException
+from utility import is_number
+
+
+def remove_whitespace(expression: str) -> str:
+    """
+    Removes whitespace from an expression after validating that it does not
+    contain two consecutive numbers.
+
+    Args:
+        expression (str): The mathematical expression.
+
+    Returns:
+        str: The expression with all whitespace removed.
+
+    Raises:
+        NumbersInARowException: If two numbers appear sequentially without an operator.
+    """
+
+    divided_by_whitespace = expression.split()
+    is_previous_number = False
+
+    for token in divided_by_whitespace:
+
+        if is_number(token):
+            if is_previous_number:
+                raise NumbersInARowException("Invalid syntax: Two numbers in a row")
+            is_previous_number = True
+
+        else:
+            is_previous_number = False
+
+    return expression.replace(' ', '')
+
+def reform_sequence_of_intensity1 (expression: str) -> str:
+    """
+        Replaces sequence of -,+ via mathematical logic:
+            even number of - becomes +
+            odd number of - becomes -
+            sequence of + becomes +
+
+        Args:
+            expression (str): The mathematical expression.
+
+        Returns:
+            str: The expression with replaced sequence of -,+ via mathematical logic.
+    """
+
+    reformed_expression = []
+    minus_count = 0
+    sequence = False
+
+    for token in expression:
+        if token == '-' or token == '+':
+            if token == '-':
+                minus_count += 1
+            sequence = True
+
+        else:
+            if sequence:
+                if minus_count % 2 == 0:
+                    reformed_expression.append("+")
+
+                else:
+                    reformed_expression.append("-")
+
+                sequence = False
+                minus_count = 0
+
+            reformed_expression.append(token)
+
+    return ''.join(reformed_expression)
+
Index: Exceptions/NumbersInARowException.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Exceptions/NumbersInARowException.py b/Exceptions/NumbersInARowException.py
new file mode 100644
--- /dev/null	(date 1766956362218)
+++ b/Exceptions/NumbersInARowException.py	(date 1766956362218)
@@ -0,0 +1,3 @@
+class NumbersInARowException(Exception):
+    """Exception raised for invalid syntax(sequence of numbers seperated by spaces)"""
+    pass
\ No newline at end of file
Index: Exceptions/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Exceptions/__init__.py b/Exceptions/__init__.py
new file mode 100644
--- /dev/null	(date 1766956362224)
+++ b/Exceptions/__init__.py	(date 1766956362224)
@@ -0,0 +1,1 @@
+from .NumbersInARowException import NumbersInARowException
\ No newline at end of file
diff --git a/Reform/__init__.py b/Reform/__init__.py
new file mode 100644
